index.js

```diff
@@ -145,6 +145,7 @@ function fastifyView (fastify, opts, next) {
 
       let compiledPage
       try {
+        options.filename = options.filename || join(templatesDir, page)
```

Comment:
I have check this PR and I don't understand this change:
With two routes:

```
fastify.get('/', (req, reply) => {
  reply.view('./templates/sample.pug', data)
})
fastify.get('/a', (req, reply) => {
  reply.view('./templates/sample2.pug', data)
})
```

the `options.filename` value will be always equal to **the first rendered page**.

This is working.. but it doesn't seem right to me.

Also this settings works:

```
options.filename = page
options.basedir = templatesDir
```
And seems better.

Consider also that using the `options` object for variable could introduce side effects, so I would use a dedicated variable instead of use it like a singleton.

Could you explain better this change or update it accordingly?

Reply:
Thing is... how do you set `options.filename` dynamically to the page that is being rendered. As I understand it, the change I did sets that option to the page being rendered after being read from the filesystem.